test/external/process_replay/process_replay.py

```diff
@@ -1,28 +1,41 @@
-#!/usr/bin/env python3
-# compare kernels created by HEAD against master
-import difflib, pickle, multiprocessing, os, logging, sqlite3, requests
-from tabulate import tabulate
-from datetime import datetime
-from typing import Dict, List, cast
-from test.external.process_replay.utils import print_diff
+import os, multiprocessing, logging, pickle, sqlite3
+from typing import Callable, List, cast
+from tinygrad.helpers import VERSION, Context, ContextVar, db_connection, getenv, tqdm
 from tinygrad.codegen.kernel import Kernel
-from tinygrad.helpers import Context, ContextVar, colored, db_connection, VERSION, getenv, tqdm
+from test.external.process_replay.utils import print_diff
 
 # *** process replay settings
+
+# internal
 PAGE_SIZE = 100
 REF = os.getenv("GITHUB_REF_NAME", "")
 MAX_DIFF_PCT = getenv("PROCESS_REPLAY_MAX_DIFF_PCT", 20)
 RUN_ID = os.getenv("GITHUB_RUN_ID", "HEAD")
 TABLE_NAME = f"process_replay_{RUN_ID}_{getenv('GITHUB_RUN_ATTEMPT')}_{VERSION}"
+early_stop = multiprocessing.Event()
+logging.basicConfig(level=logging.INFO, format='%(message)s')
```

Comment:
consider setting proper levels instead of changing behavior based on `ASSERT_DIFF`. it's easier to maintain if the log flows are the same, and the only difference is that warning does throw

Reply:
hmm, I wonder why changing behavior based on ASSERT_DIFF is bad, I think it's intentional.
but yea logging is exactly the same, whether CI is red or not is different.